Fix issue with DisplacementAboutAxis with finite rotation

bwspenc opened the pull request:
closes #15092

The calculations are set up to rotate from the original configuration,
but it has 'use_displaced_mesh=true' set. Switch it to 'false'.

bwspenc committed:
Fix issue with DisplacementAboutAxis with finite rotation closes #15092

The calculations are set up to rotate from the original configuration,
but it has 'use_displaced_mesh=true' set. Switch it to 'false'.

Modified one of the existing tests to apply a 90 degree rigid body
rotation to an element to verify that this is behaving correctly now.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/506336/) on 326a7af wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/15093/site)

This comment will be updated on new commits.

sapitts commented on modules/tensor_mechanics/test/tests/torque_reaction/disp_about_axis_axial_motion.i:
```diff
@@ -132,35 +43,48 @@
   [../]
-
-# Because want to keep the rotation fixed about the z axis,
-# do not apply a DisplacementAboutAxis BC on the disp_z
```
I'd like to keep these two lines of comments in the input file --this very explanation actually helped me out a bunch when I was setting up the torsion tube simulation

bwspenc replied:
Sure, I can add them back.  I thought they were confusing, myself!

sapitts replied:
If there's a better way to word the comments, please change it to make it less confusing. I had found the 'captain obvious' comments helpful when I was trying to figure out if I should or should not apply a `disp_z` boundary condition

bwspenc replied:
Note that I revised this comment and moved it above this set of BCs. It actually is OK to specify those in the z direction, but it would give the same result in this case since there is no deformation. It would be equivalent to applying a DirichletBC in the z direction.

sapitts commented on modules/tensor_mechanics/test/tests/torque_reaction/disp_about_axis_axial_motion.i:
```diff
@@ -176,24 +100,17 @@
+  dtmin = 1 # die instead of cutting the timestep
 
-  end_time = 0.5
+  end_time = 90
```
This test still remains under the two second mark with this large of an end_time?

bwspenc replied:
Yes, it runs fast. I'll verify, but I'm pretty sure its well under a second.

bwspenc replied:
It took 1.5 s.  I doubled the time step size, and now it runs in .677s.

sapitts commented on modules/tensor_mechanics/test/tests/torque_reaction/tests:
```diff
@@ -1,34 +1,53 @@
   [./torque_reaction_2D]
+    design = TorqueReaction.md
+    issues = '#6484'
+    requirement = 'The TorqueReaction Postprocessor shall compute the torque for an applied deformation at the end of a beam in 2D'
```
Are these supposed to all start with 'The system shall.....'? I left my SQA notes in the office

bwspenc replied:
Yeah, I wasn't sure. I think you're right.

sapitts commented on modules/tensor_mechanics/test/tests/torque_reaction/tests:
```diff
@@ -1,34 +1,53 @@
     abs_zero = 1.e-8
   [../]
   [./torque_reaction_cylinder]
+    design = 'TorqueReaction.md DisplacementAboutAxis.md'
```
Any chance you'd be willing to rename the input/gold file pairs from the previous two tests to remove the `_tm`?

bwspenc replied:
I thought the same thing, and was just being lazy.  Sure, I'll fix that. Then you're going to want me to convert them all to use the Action too...

sapitts commented on modules/tensor_mechanics/test/tests/torque_reaction/tests:
```diff
@@ -1,34 +1,53 @@
+    requirement = 'The TorqueReaction Postprocessor shall compute the torque for an applied deformation at the end of a beam in 2D'
     type = 'Exodiff'
     input = 'torque_reaction_tm.i'
     exodiff = 'torque_reaction_tm_out.e'
```
Does the regolded exodus file still match the analytical solution that is given in the input file / does that analytical solution need to be updated to reflect the increased applied rotation displacement?

bwspenc replied:
This test was actually not affected by the change because it doesn't use the `DisplacementAboutAxis` BC. The `torque_reaction_cylinder` test was slightly affected, but it did not list a result in the .i file, and the change was very minor because it was in the small rotation regime.

sapitts reviewed (changes requested):
Looks quite close to me; please double check on the validity of the analytical solutions in the torque reaction input files and on the proper wording for the SQA requirements (I'm afraid we'll forget these if we don't double check it now)

bwspenc committed:
Address review comments closes #15092

bwspenc committed:
Edits to renamed files to address review comments closes #15092

bwspenc committed:
Convert tests to use TM master action closes #15092

Note that the torque_reaction_3D test had to be re-golded because
it was previously set up in an inconsistent way. The kernels were
running on the displaced mesh, but the model was using a small strain
calculator.

bwspenc commented:
@sapitts I think I've addressed all the issues you raised. I also converted all the tests in this directory to use the TM Master action. That made one of the tests diff (torque_reaction_3D) because it was set up in an inconsistent way -- the Kernel was running on the displaced mesh, but it used a small strain calculator, so I had to update the gold file.

sapitts reviewed (dismissed):
Looks good to me; thanks for making these changes/ checks. 
@aeslaughter would you give this a final review and merge please?

bwspenc commented:
I just rebased and pushed (without changing anything) to trigger the tests to re-run now that the other unrelated issues that were making the tests fail have hopefully been addressed.

bwspenc committed:
Update documentation and use tagging system closes #15092

bwspenc commented:
Pesky documentation -- I broke it multiple places because of changes to those test input files. I switched from `save_in` in this last round since the doc mentioned it, but it's deprecated. Should be good now.

bwspenc commented:
OK, looks like this finally passed. @aeslaughter, if you could take a look at this, I'd appreciate it.
